core/src/main/java/org/eclipse/hono/adapter/AdapterConfig.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) 2016 Red Hat
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Red Hat - initial creation
+ */
+
+package org.eclipse.hono.adapter;
+
+import io.vertx.core.Vertx;
+import org.eclipse.hono.config.HonoClientConfigProperties;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Minimum configuration for protocol adapters
+ */
+@Configuration
+public abstract class AdapterConfig {
```

Comment:
You are right on that, it's not necessary that it should be abstract absolutely. Now I remember that without "abstracting" it, the Hono Server application tries to use even this file as configuration file. It throws the following exception : 

09:26:44.633 [main] ERROR o.s.boot.SpringApplication - Application startup failed
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'application': Unsatisfied dependency expressed through method 'setVertx' parameter 0: No qualifying bean of type [io.vertx.core.Vertx] is defined: expected single matching bean but found 2: getInstance,getVertx; nested exception is org.springframework.beans.factory.NoUniqueBeanDefinitionException: No qualifying bean of type [io.vertx.core.Vertx] is defined: expected single matching bean but found 2: getInstance,getVertx

because the ApplicationConfig in Hono already has getInstance() which returns a Vert.x.

How can we avoid this ? We have AdapterConfig in the "core" but it should be excluded when Hono server application starts. We need this config only for adapters.